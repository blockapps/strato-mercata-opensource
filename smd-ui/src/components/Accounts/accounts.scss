.accounts-margin-top {
  bottom: 10px;
  padding-right: 7px;
  padding-left: 10px;
  overflow-y: auto;
  top: 170px;
}

.main-div {
  overflow-y: auto;
  margin-top: 40px;
  height: 70vh;
}

.main-div .smd-margin-8:first-of-type {
  margin-top: 0px;
}

.main-div::-webkit-scrollbar {
  width: 10px;
}

.account-details {
  margin-top: 40px;
  overflow-y: auto;
  height: 70vh;
  overflow-x: hidden;
}

.account-details::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
  border-radius: 4px;
  background: #2F424D; 
}
 
::-webkit-scrollbar-thumb {
  border-radius: 4px; 
  background: #F4F8FA; 
}

::-webkit-scrollbar-thumb:hover {
  background: #BBCBD4; 
}

.send-tokens-dialog .chain-field {
  width: 100%;
}