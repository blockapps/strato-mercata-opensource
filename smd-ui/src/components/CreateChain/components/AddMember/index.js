import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Button, Dialog, Intent } from '@blueprintjs/core';

import mixpanelWrapper from '../../../../lib/mixpanelWrapper';
import { Field, reduxForm } from 'redux-form';
import { openAddMemberModal, closeAddMemberModal } from '../../createChain.actions';
import { validate } from './validate';

class AddMember extends Component {

  constructor(props) {
    super(props)
    this.state = {
      form: {
      },
      orgName: ``,
      orgUnit: ``,
      commonName: ``,
      access: true,
      errors: null
    }
  }

  closeModal = () => {
    this.props.closeAddMemberModal();
  }

  componentDidMount() {
    mixpanelWrapper.track("add_member_loaded");
  }

  handleOrgNameChange(event) {
    this.setState({
      orgName: event.target.value
    });
  }

  handleOrgUnitChange(event) {
    this.setState({
      orgUnit: event.target.value
    });
  }

  handleCommonNameChange(event) {
    this.setState({
      commonName: event.target.value
    });
  }

  handleAccessChange(event) {
    this.setState({
      access: event.target.value
    });
  }

  errorMessageFor(fieldName) {
    if (this.state.errors && this.state.errors[fieldName]) {
      return this.state.errors[fieldName];
    }
    return null;
  }

  submit = () => {
    let data = {
      orgName: this.state.orgName,
      orgUnit: this.state.orgUnit,
      commonName: this.state.commonName,
      access: this.state.access
    }
    
    let errors = validate(data);
    this.setState({ errors });

    if (!Object.values(errors).length) {
      mixpanelWrapper.track('add_member_submit_click_successful');
      this.props.handler(data);
      this.props.reset();
      this.closeModal();
    }
  }

  render() {
    return (
      <div >
        <Button onClick={() => {
          mixpanelWrapper.track("add_member_open_click");
          this.props.openAddMemberModal();
        }} className="pt-intent-primary pt-icon-add"
          style={{ marginTop: '8px' }}
          text="Add Member" />

        <Dialog
          iconName="add"
          isOpen={this.props.isOpen}
          onClose={this.closeModal}
          title="Add Member"
          className="pt-dark"
        >
          <form>
            <div className="pt-dialog-body">

              {
                // !this.state.form.userSelected &&
                <div className="row">
                  <div className="col-sm-3 text-right">
                    <label className="pt-label smd-pad-4">
                      Org Name:
                  </label>
                  </div>
                  <div className="col-sm-9 smd-pad-4">
                    <Field
                      id="orgName"
                      className="form-width pt-input"
                      placeholder="Org Name"
                      name="orgName"
                      component="input"
                      dir="auto"
                      title="orgName"
                      onChange={(e) => this.handleOrgNameChange(e)}
                      required
                    />
                    <br /><span className="error-text">{this.errorMessageFor('orgName')}</span>
                  </div>
                </div>
              }

              <div className="row">
                <div className="col-sm-3 text-right">
                  <label className="pt-label smd-pad-4">
                    Org Unit:
                  </label>
                </div>
                <div className="col-sm-9 smd-pad-4">
                  <div className="form-width">
                    <Field
                      name="orgUnit"
                      component="input"
                      type="text"
                      placeholder="Org Unit"
                      value={this.state.orgUnit}
                      className="pt-input form-width"
                      onChange={(e) => this.handleOrgUnitChange(e)}
                      required
                    />
                    <br /><span className="error-text">{this.errorMessageFor('orgUnit')}</span>
                  </div>
                </div>
              </div>

              <div className="row">
                <div className="col-sm-3 text-right">
                  <label className="pt-label smd-pad-4">
                    Common Name:
                  </label>
                </div>
                <div className="col-sm-9 smd-pad-4">
                  <div className="form-width">
                    <Field
                      name="commonName"
                      component="input"
                      type="text"
                      placeholder="Common Name"
                      value={this.state.commonName}
                      className="pt-input form-width"
                      onChange={(e) => this.handleCommonNameChange(e)}
                      required
                    />
                    <br /><span className="error-text">{this.errorMessageFor('commonName')}</span>
                  </div>
                </div>
              </div>
              <div className="row">
                <div className="col-sm-3 text-right">
                  <label className="pt-label smd-pad-4">
                    Access
                  </label>
                </div>
                <div className="col-sm-9 smd-pad-4">
                  <div className="form-width">
                  <Field
                      style={{ marginLeft: 25 }}
                      name="radio"
                      component="input"
                      type="radio"
                      value={true}
                      label='Add'
                      checked={this.state.access === true}
                      onClick={() => {
                        this.setState((prevState) => {
                          return {access:true };
                        });
                      }}
                    /> Add
                  <Field
                      style={{ marginLeft: 25 }}
                      name="radio"
                      component="input"
                      type="radio"
                      value={false}
                      label='Remove'
                      checked={this.state.access === false}
                      onClick={() => {
                        this.setState((prevState) => {
                          return {access:false };
                        });
                      }}
                    /> Remove
                    <br /><span className="error-text">{this.errorMessageFor('access')}</span>
                  </div>
                </div>
              </div>
            </div>

            <div className="pt-dialog-footer">
              <div className="pt-dialog-footer-actions">
                <Button text="Cancel" onClick={() => {
                  mixpanelWrapper.track("add_member_cancel");
                  this.props.reset();
                  this.closeModal();
                }} />
                <Button
                  intent={Intent.PRIMARY}
                  onClick={this.submit}
                  text="Add Member"
                />
              </div>
            </div>
          </form>
        </Dialog>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    isOpen: state.createChain.isAddMemberModalOpen,
    initialValues: {
      orgName: "",
      orgUnit: "",
      commonName: "",
      access: true
    }
  };
}

const formed = reduxForm({ form: 'add-member' })(AddMember);
const connected = connect(mapStateToProps, {
  openAddMemberModal,
  closeAddMemberModal
})(formed);

export default withRouter(connected);
