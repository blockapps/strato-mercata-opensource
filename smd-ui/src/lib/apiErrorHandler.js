export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
