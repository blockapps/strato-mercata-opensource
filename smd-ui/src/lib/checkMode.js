import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}