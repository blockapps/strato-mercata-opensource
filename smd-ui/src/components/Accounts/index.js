import React, { Component } from 'react';
import {
  fetchAccounts,
  fetchOauthAccounts,
} from './accounts.actions';
import mixpanelWrapper from '../../lib/mixpanelWrapper';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Tour from '../Tour';
import BlocAccounts from './components/BlocAccounts';
import OauthAccounts from './components/OauthAccounts';
import './accounts.css';
import { isOauthEnabled } from '../../lib/checkMode';
import ReactGA from 'react-ga4';

const tourSteps = [/* {
    title: 'Create User',
    text: 'Create a user here',
    selector: '#accounts-create-user-button',
    position: 'bottom', type: 'hover',
    isFixed: true,
  }, ) */
  {
    title: 'Upload a Smart Contract',
    text: 'Drag and drop a <strong>.sol</strong> file, and you will be able to manage your ' +
      'Smart Contract from within the STRATO dashboard.',
    /* text: '<div class="inline-code-sample">contract RentSplit {<br>address <strong>Roommate 1</strong>;<br><strong>Roommate 2</strong>;<br><strong>Roommate 3</strong>;<br>mapping (address => uint) RentSplit;<br></div>', */
    selector: '#contracts',
    position: 'bottom',
    isFixed: true
  }
];

class Accounts extends Component {

  constructor() {
    super()
    this.state = {
      selected: 0
    }
  }

  componentDidMount() {
    if (isOauthEnabled()) {
      this.props.fetchOauthAccounts();
    } else {
      this.props.fetchAccounts(true, true, this.props.selectedChain);
    }
    mixpanelWrapper.track('accounts_page_load');
    ReactGA.send({hitType: "pageview", page: "/accounts", title: "Accounts"});
  }

  render() {
    return (
      <div className="container-fluid pt-dark">
        <Tour
          name="accounts"
          steps={tourSteps}
          finalStepSelector='#contracts'
          nextPage='chains' />
        {isOauthEnabled() ? <OauthAccounts /> : <BlocAccounts />}
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {};
}

export default withRouter(
  connect(mapStateToProps,
    {
      fetchOauthAccounts,
      fetchAccounts
    }
  )(Accounts));
