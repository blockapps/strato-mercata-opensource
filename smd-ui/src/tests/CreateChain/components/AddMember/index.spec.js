import React from 'react';
import AddMember, { mapStateToProps } from '../../../../components/CreateChain/components/AddMember';
import { reducer as formReducer } from 'redux-form';
import { createStore, combineReducers } from 'redux';
import * as checkMode from '../../../../lib/checkMode';
import { indexAccountsMock } from '../../../Accounts/accountsMock';

describe('CreateChain: index', () => {

  let store;

  beforeEach(() => {
    store = createStore(combineReducers({ form: formReducer }))
  });

  describe('render component (non Oauth mode)', () => {

    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
    })

    test('with empty values', () => {
      const props = {
        initialValues: {
        },
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        initialValues: {
        },
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('form fields', () => {
      const props = {
        initialValues: {
        },
        handler: jest.fn(),
        reset: jest.fn(),
        closeAddMemberModal: jest.fn(),
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.state()).toMatchSnapshot();

      // on address radio selection
      wrapper.find('Field').first().simulate('click');
      wrapper.find('Field').at(1).simulate('change', { target: { value: 'BlockApps' } });
      wrapper.find('Field').at(2).simulate('change', { target: { value: 'Engineering' } });
      wrapper.find('Field').at(3).simulate('change', { target: { value: 'James Hormuzdiar' } });
      wrapper.find('Field').at(4).simulate('change', { target: { value: true } });
      expect(wrapper.state()).toMatchSnapshot();

      // on user radio selection
      wrapper.find('Field').at(1).simulate('click');
      wrapper.find('Field').at(1).simulate('change', { target: { value: 'BlockApps' } });
      wrapper.find('Field').at(2).simulate('change', { target: { value: 'Engineering' } });
      wrapper.find('Field').at(3).simulate('change', { target: { value: 'James Hormuzdiar' } });
      wrapper.find('Field').at(4).simulate('change', { target: { value: false } });
      // console.log(wrapper.state());
      expect(wrapper.state()).toMatchSnapshot();

      wrapper.find('Button').last().simulate('click');
      expect(props.handler).toHaveBeenCalled();
      expect(props.handler).toHaveBeenCalledTimes(1);
      expect(props.closeAddMemberModal).toHaveBeenCalled();
      expect(props.closeAddMemberModal).toHaveBeenCalledTimes(1);
    });

    test('errorMessageFor', () => {
      const props = {
        initialValues: {
        },
        handler: jest.fn(),
        reset: jest.fn(),
        closeAddMemberModal: jest.fn(),
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.instance().errorMessageFor('orgName')).toMatchSnapshot();
      wrapper.setState({ errors: { username: 'required' } })
      expect(wrapper.instance().errorMessageFor('orgName')).toMatchSnapshot()
    });

  });

  describe('render component (Oauth mode)', () => {

    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(true);
    })

    test('with empty values', () => {
      const props = {
        initialValues: {
        },
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        initialValues: {
        },
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      expect(wrapper.debug()).toMatchSnapshot();
    });

    describe('form fields', () => {

      test('with values', () => {
        const props = {
          isOpen: true,
          initialValues: {
          },
          handler: jest.fn(),
          reset: jest.fn(),
          closeAddMemberModal: jest.fn(),
          store: store
        };

        const wrapper = shallow(
          <AddMember.WrappedComponent {...props} />
        ).dive().dive().dive();

        expect(wrapper.state()).toMatchSnapshot();
        wrapper.find('Field').at(1).simulate('change', { target: { value: 'BlockApps' } });
        wrapper.find('Field').at(2).simulate('change', { target: { value: 'Engineering' } });
        wrapper.find('Field').at(3).simulate('change', { target: { value: 'James Hormuzdiar' } });
        expect(wrapper.state()).toMatchSnapshot();
        wrapper.find('Button').last().simulate('click');
        expect(props.handler).toHaveBeenCalled();
        expect(props.handler).toHaveBeenCalledTimes(1);
        expect(props.closeAddMemberModal).toHaveBeenCalled();
        expect(props.closeAddMemberModal).toHaveBeenCalledTimes(1);
      });

      test('without values', () => {
        const props = {
          initialValues: {
          },
          handler: jest.fn(),
          reset: jest.fn(),
          closeAddMemberModal: jest.fn(),
          store: store
        };

        const wrapper = shallow(
          <AddMember.WrappedComponent {...props} />
        ).dive().dive().dive();
        wrapper.find('Field').at(2).simulate('change', { target: { value: '' } });
        wrapper.find('Button').last().simulate('click');
        expect(props.handler).toHaveBeenCalled();
        expect(props.handler).toHaveBeenCalledTimes(1);
        expect(props.closeAddMemberModal).toHaveBeenCalled();
        expect(props.closeAddMemberModal).toHaveBeenCalledTimes(1);
      });

    });

  });

  describe('add member modal', () => {

    test('open modal', () => {
      const props = {
        initialValues: {
        },
        handler: jest.fn(),
        reset: jest.fn(),
        openAddMemberModal: jest.fn(),
        closeAddMemberModal: jest.fn(),
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      wrapper.find('Button').first().simulate('click');
      expect(props.openAddMemberModal).toHaveBeenCalled();
      expect(props.openAddMemberModal).toHaveBeenCalledTimes(1);
    });

    test('close modal', () => {
      const props = {
        initialValues: {
        },
        handler: jest.fn(),
        reset: jest.fn(),
        openAddMemberModal: jest.fn(),
        closeAddMemberModal: jest.fn(),
        store: store
      };

      const wrapper = shallow(
        <AddMember.WrappedComponent {...props} />
      ).dive().dive().dive();

      wrapper.find('Button').at(1).simulate('click');
      expect(props.closeAddMemberModal).toHaveBeenCalled();
      expect(props.closeAddMemberModal).toHaveBeenCalledTimes(1);
    });

  });

  test('mapStateToProps with default state', () => {
    const state = {
      createChain: {
        isAddMemberModalOpen: true
      },
    }

    expect(mapStateToProps(state)).toMatchSnapshot();
  });

});