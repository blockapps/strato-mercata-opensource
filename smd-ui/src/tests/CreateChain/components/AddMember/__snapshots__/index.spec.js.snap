// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChain: index mapStateToProps with default state 1`] = `
Object {
  "initialValues": Object {
    "access": true,
    "commonName": "",
    "orgName": "",
    "orgUnit": "",
  },
  "isOpen": true,
}
`;

exports[`CreateChain: index render component (Oauth mode) form fields with values 1`] = `
Object {
  "access": true,
  "commonName": "",
  "errors": null,
  "form": Object {},
  "orgName": "",
  "orgUnit": "",
}
`;

exports[`CreateChain: index render component (Oauth mode) form fields with values 2`] = `
Object {
  "access": true,
  "commonName": "Engineering",
  "errors": null,
  "form": Object {},
  "orgName": "",
  "orgUnit": "BlockApps",
}
`;

exports[`CreateChain: index render component (Oauth mode) with empty values 1`] = `
"<div>
  <Blueprint.Button onClick={[Function: onClick]} className=\\"pt-intent-primary pt-icon-add\\" style={{...}} text=\\"Add Member\\" />
  <Blueprint.Dialog iconName=\\"add\\" isOpen={false} onClose={[Function]} title=\\"Add Member\\" className=\\"pt-dark\\" canOutsideClickClose={true}>
    <form>
      <div className=\\"pt-dialog-body\\">
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <Field id=\\"orgName\\" className=\\"form-width pt-input\\" placeholder=\\"Org Name\\" name=\\"orgName\\" component=\\"input\\" dir=\\"auto\\" title=\\"orgName\\" onChange={[Function: onChange]} required={true} />
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Unit:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"orgUnit\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Org Unit\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Common Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"commonName\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Common Name\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Access
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={true} label=\\"Add\\" checked={true} onClick={[Function: onClick]} />
               Add
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={false} label=\\"Remove\\" checked={false} onClick={[Function: onClick]} />
               Remove
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
      </div>
      <div className=\\"pt-dialog-footer\\">
        <div className=\\"pt-dialog-footer-actions\\">
          <Blueprint.Button text=\\"Cancel\\" onClick={[Function: onClick]} />
          <Blueprint.Button intent={0} onClick={[Function]} text=\\"Add Member\\" />
        </div>
      </div>
    </form>
  </Blueprint.Dialog>
</div>"
`;

exports[`CreateChain: index render component (Oauth mode) with values 1`] = `
"<div>
  <Blueprint.Button onClick={[Function: onClick]} className=\\"pt-intent-primary pt-icon-add\\" style={{...}} text=\\"Add Member\\" />
  <Blueprint.Dialog iconName=\\"add\\" isOpen={false} onClose={[Function]} title=\\"Add Member\\" className=\\"pt-dark\\" canOutsideClickClose={true}>
    <form>
      <div className=\\"pt-dialog-body\\">
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <Field id=\\"orgName\\" className=\\"form-width pt-input\\" placeholder=\\"Org Name\\" name=\\"orgName\\" component=\\"input\\" dir=\\"auto\\" title=\\"orgName\\" onChange={[Function: onChange]} required={true} />
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Unit:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"orgUnit\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Org Unit\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Common Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"commonName\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Common Name\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Access
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={true} label=\\"Add\\" checked={true} onClick={[Function: onClick]} />
               Add
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={false} label=\\"Remove\\" checked={false} onClick={[Function: onClick]} />
               Remove
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
      </div>
      <div className=\\"pt-dialog-footer\\">
        <div className=\\"pt-dialog-footer-actions\\">
          <Blueprint.Button text=\\"Cancel\\" onClick={[Function: onClick]} />
          <Blueprint.Button intent={0} onClick={[Function]} text=\\"Add Member\\" />
        </div>
      </div>
    </form>
  </Blueprint.Dialog>
</div>"
`;

exports[`CreateChain: index render component (non Oauth mode) errorMessageFor 1`] = `null`;

exports[`CreateChain: index render component (non Oauth mode) errorMessageFor 2`] = `null`;

exports[`CreateChain: index render component (non Oauth mode) form fields 1`] = `
Object {
  "access": true,
  "commonName": "",
  "errors": null,
  "form": Object {},
  "orgName": "",
  "orgUnit": "",
}
`;

exports[`CreateChain: index render component (non Oauth mode) form fields 2`] = `
Object {
  "access": true,
  "commonName": "Engineering",
  "errors": null,
  "form": Object {},
  "orgName": "",
  "orgUnit": "BlockApps",
}
`;

exports[`CreateChain: index render component (non Oauth mode) form fields 3`] = `
Object {
  "access": true,
  "commonName": "Engineering",
  "errors": null,
  "form": Object {},
  "orgName": "",
  "orgUnit": "BlockApps",
}
`;

exports[`CreateChain: index render component (non Oauth mode) with empty values 1`] = `
"<div>
  <Blueprint.Button onClick={[Function: onClick]} className=\\"pt-intent-primary pt-icon-add\\" style={{...}} text=\\"Add Member\\" />
  <Blueprint.Dialog iconName=\\"add\\" isOpen={false} onClose={[Function]} title=\\"Add Member\\" className=\\"pt-dark\\" canOutsideClickClose={true}>
    <form>
      <div className=\\"pt-dialog-body\\">
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <Field id=\\"orgName\\" className=\\"form-width pt-input\\" placeholder=\\"Org Name\\" name=\\"orgName\\" component=\\"input\\" dir=\\"auto\\" title=\\"orgName\\" onChange={[Function: onChange]} required={true} />
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Unit:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"orgUnit\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Org Unit\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Common Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"commonName\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Common Name\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Access
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={true} label=\\"Add\\" checked={true} onClick={[Function: onClick]} />
               Add
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={false} label=\\"Remove\\" checked={false} onClick={[Function: onClick]} />
               Remove
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
      </div>
      <div className=\\"pt-dialog-footer\\">
        <div className=\\"pt-dialog-footer-actions\\">
          <Blueprint.Button text=\\"Cancel\\" onClick={[Function: onClick]} />
          <Blueprint.Button intent={0} onClick={[Function]} text=\\"Add Member\\" />
        </div>
      </div>
    </form>
  </Blueprint.Dialog>
</div>"
`;

exports[`CreateChain: index render component (non Oauth mode) with values 1`] = `
"<div>
  <Blueprint.Button onClick={[Function: onClick]} className=\\"pt-intent-primary pt-icon-add\\" style={{...}} text=\\"Add Member\\" />
  <Blueprint.Dialog iconName=\\"add\\" isOpen={false} onClose={[Function]} title=\\"Add Member\\" className=\\"pt-dark\\" canOutsideClickClose={true}>
    <form>
      <div className=\\"pt-dialog-body\\">
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <Field id=\\"orgName\\" className=\\"form-width pt-input\\" placeholder=\\"Org Name\\" name=\\"orgName\\" component=\\"input\\" dir=\\"auto\\" title=\\"orgName\\" onChange={[Function: onChange]} required={true} />
            <br />
            <span className=\\"error-text\\" />
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Org Unit:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"orgUnit\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Org Unit\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Common Name:
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field name=\\"commonName\\" component=\\"input\\" type=\\"text\\" placeholder=\\"Common Name\\" value=\\"\\" className=\\"pt-input form-width\\" onChange={[Function: onChange]} required={true} />
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
        <div className=\\"row\\">
          <div className=\\"col-sm-3 text-right\\">
            <label className=\\"pt-label smd-pad-4\\">
              Access
            </label>
          </div>
          <div className=\\"col-sm-9 smd-pad-4\\">
            <div className=\\"form-width\\">
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={true} label=\\"Add\\" checked={true} onClick={[Function: onClick]} />
               Add
              <Field style={{...}} name=\\"radio\\" component=\\"input\\" type=\\"radio\\" value={false} label=\\"Remove\\" checked={false} onClick={[Function: onClick]} />
               Remove
              <br />
              <span className=\\"error-text\\" />
            </div>
          </div>
        </div>
      </div>
      <div className=\\"pt-dialog-footer\\">
        <div className=\\"pt-dialog-footer-actions\\">
          <Blueprint.Button text=\\"Cancel\\" onClick={[Function: onClick]} />
          <Blueprint.Button intent={0} onClick={[Function]} text=\\"Add Member\\" />
        </div>
      </div>
    </form>
  </Blueprint.Dialog>
</div>"
`;
